eva_submission/eload_ingestion.py

```diff
@@ -114,3 +152,215 @@ def load_from_ena(self):
+                    vcf_path=vcf_path,
+                    output_vcf=output_vcf,
+                    mongo_host=mongo_host,
+                    mongo_user=mongo_user,
+                    mongo_pass=mongo_pass,
+                    postgres_url=pg_url,
+                    postgres_user=pg_user,
+                    postgres_pass=pg_pass
+                ))
+            prop_files.append(str(properties_filename))
+        return prop_files
+
+    def create_variant_load_properties(self):
+        """
+        Creates properties files for the variant load pipeline, one for each VCF file.
+        """
+        prop_files = []
+        for vcf_path in self.vcfs_to_load:
+            filename = vcf_path.stem
+            properties_filename = self.project_dir.joinpath(f'load_{filename}.properties')
+            with open(properties_filename, 'w+') as f:
+                f.write(variant_load_props_template(
+                    project_accession=self.project_accession,
+                    analysis_accession=self.eload_cfg.query('brokering', 'ena', 'ANALYSIS'),
+                    vcf_path=vcf_path,
+                    aggregation=self.eload_cfg.query(self.config_section, 'aggregation'),
+                    study_name=self.get_study_name(),
+                    fasta=self.eload_cfg.query('submission', 'assembly_fasta'),
+                    output_dir=self.project_dir.joinpath(project_dirs['transformed']),
+                    annotation_dir=self.project_dir.joinpath(project_dirs['annotation']),
+                    stats_dir=self.project_dir.joinpath(project_dirs['stats']),
+                    db_name=self.eload_cfg.query(self.config_section, 'database', 'db_name'),
+                    vep_species=self.get_vep_species(),
+                    vep_version=self.eload_cfg.query(self.config_section, 'variant_load', 'vep', 'version'),
+                    vep_cache_version=self.eload_cfg.query(self.config_section, 'variant_load', 'vep', 'cache_version')
+                ))
+            prop_files.append(str(properties_filename))
+        return prop_files
+
+    def get_mongo_creds(self):
+        properties = get_properties_from_xml_file('production', self.settings_xml_file)
+        mongo_host = str(str(properties['eva.mongo.host']).split(',')[0]).split(':')[0]
+        mongo_user = properties['eva.mongo.user']
+        mongo_pass = properties['eva.mongo.passwd']
+        return mongo_host, mongo_user, mongo_pass
+
+    def get_pg_creds(self):
+        properties = get_properties_from_xml_file('development', self.settings_xml_file)
+        pg_url = properties['eva.evapro.jdbc.url']
+        pg_user = properties['eva.evapro.user']
+        pg_pass = properties['eva.evapro.password']
+        return pg_url, pg_user, pg_pass
+
+    def get_pg_conn(self):
+        pg_url, pg_user, pg_pass = self.get_pg_creds()
+        # need to cut the jdbc: from the front of the pg_url
+        return psycopg2.connect(pg_url[5:], user=pg_user, password=pg_pass)
```

Comment:
Please investigate the urllib.parse [urlsplit](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit) to do this in a standardized way.